frontend/packages/topology/src/components/export-app/__tests__/ExportViewLogButton.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { JobModel, PodModel } from '@console/internal/models';
+import store from '@console/internal/redux';
+import ExportViewLogButton from '../ExportViewLogButton';
+
+jest.mock('@console/internal/components/utils/k8s-watch-hook', () => ({
+  useK8sWatchResource: jest.fn(),
+}));
+
+const mockK8sWatchResource = useK8sWatchResource as jest.Mock;
+type Component = typeof ExportViewLogButton;
+type Props = React.ComponentProps<Component>;
+let ViewLogButtonProps: Props;
```

Comment:
IMHO this type "extraction" makes the complete test less readable and is not necessary, you can just add namespace="test" to your react elements below.

I recommend to drop this:

```suggestion
```

Reply:
removed